packages/core/src/components/MonthPicker/MonthPicker.jsx

```diff
@@ -0,0 +1,220 @@
+
+    if (typeof this.props.selectedMonths === 'undefined') {
+      this.isControlled = false;
+      // Since this isn't a controlled component, we need a way
+      // to track when the value has changed.
+      this.state = {
+        selectedMonths:
+          this.props.defaultSelectedMonths || this.monthsArray(false)
+      };
+    } else {
+      this.isControlled = true;
+    }
+  }
+
+  selectedMonths() {
+    if (this.isControlled) {
+      return this.props.selectedMonths;
+    }
+
+    return this.state.selectedMonths;
+  }
+
+  disabledMonths() {
+    return this.props.disabledMonths || this.monthsArray(false);
+  }
+
+  /**
+   * Generates an array of 12 booleans with the given value
+   *
+   * @param  {boolean} [selected] value to populate array items
+   * @return {boolean[]}          array of 12 booleans
+   */
+  monthsArray(selected) {
+    const array = [];
+    for (let i = 0; i < NUM_MONTHS; i++) {
+      array[i] = selected;
+    }
+    return array;
+  }
+
+  handleChange(event) {
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+
+    if (!this.isControlled) {
+      const index = event.target.value;
+      const selectedMonths = this.state.selectedMonths.slice();
+      selectedMonths[index] = !selectedMonths[index];
+      this.setState({ selectedMonths });
+    }
+  }
+
+  handleSelectAll() {
+    if (this.props.onSelectAll) {
+      this.props.onSelectAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({
+        selectedMonths: this.disabledMonths().map(disabled => !disabled)
+      });
+    }
+  }
+
+  handleClearAll() {
+    if (this.props.onClearAll) {
+      this.props.onClearAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({ selectedMonths: this.monthsArray(false) });
+    }
+  }
+
+  renderMonths(selectedMonths) {
+    const disabledMonths = this.disabledMonths();
+    const { name, inversed } = this.props;
+    return (
+      <ul className="ds-c-list--bare ds-u-display--flex ds-u-justify-content--between ds-u-flex-wrap--wrap">
+        {this.months.map((month, i) => (
+          <Choice
+            name={name}
+            value={i}
+            checked={selectedMonths[i]}
+            key={month}
+            onChange={e => this.handleChange(e)}
+            className="ds-c-month-picker__month"
+            disabled={disabledMonths[i]}
+            inversed={inversed}
+            aria-label={this.monthsLong[i]}
+          >
+            {month}
+          </Choice>
+        ))}
+      </ul>
+    );
+  }
+
+  renderButton(text, onClick) {
+    return (
+      <Button
+        variation="primary"
+        size="small"
+        className="ds-u-margin-right--1"
+        onClick={onClick}
+        inversed={this.props.inversed}
+      >
+        {text}
+      </Button>
+    );
+  }
+
+  render() {
+    const { selectAllText, clearAllText } = this.props;
+    return (
+      <div className="ds-c-month-picker ds-u-margin-y--3">
```

Comment:
Should this component also render the field's label, like we do with `DateField`? Semantically, these choices would be wrapped in a `fieldset` and the label would be rendered as a `legend`.

Reply:
@cindyphan, thoughts on having a visible legend/label?
@sawyerh @1Copenut significance of having a screen-reader-only legend?

Reply:
You would need a label, no? Otherwise how would a user know what the months are being asked for? 😅

Reply:
Instead of a legend, I would opt for a descriptive header that the UL is described by. Something like:
`<h4 id="month-checklist-header">Select months for your situation</h4>
  <ul
    aria-labelledby="month-checklist-header"
    className="ds-c-list--bare ds-u-display--flex ds-u-justify-content--between ds-u-flex-wrap--wrap"
  >`

This gives the list a nice header that is read out when the first list item is focused.

Reply:
I lean more toward a `hN` than a `legend` because this may or may not be contained in a `fieldset`. The months grid could be in a div, or other component, so having a descriptive header that screen readers can hook into via the navigate by header or header rotor seems the best fit.

Reply:
Reading this thread a little more closely, I'll qualify my previous comment. I'd still like to see a descriptive header, that could be a legend, presented visually. 

The header/legend could also contain some screen reader only text wrapped in a span, describing how to use the grid's select all and clear all features.

Reply:
Good discussion 👍 I mentioned wrapping it in a `fieldset` since I believe this is what you should typically use to wrap `input`'s and `label`'s ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset)). This is also [how the ChoiceList component is setup](https://github.com/CMSgov/design-system/blob/master/packages/core/src/components/ChoiceList/ChoiceList.jsx#L128-L143), and the `MonthPicker` is basically a `ChoiceList` that is layed out differently.

Reply:
I did some quick screen reader testing with OSX + VoiceOver. Here's what I found:
* The screen reader only legend text is read every time a checkbox receives focus. This happens because the divs are treated as unique entities, and not part of a list.
* The legend is only read once if the checkboxes are grouped as part of a list, wrapped in list items like on the WDS example: https://standards.usa.gov/components/form-controls/#checkboxes

Reply:
I haven't evaluated this for NVDA or JAWS, but my thinking is they will behave in same/similar fashion. The use of divs over a list for the rendered output wrappers might be a separate discussion; I only brought it up here because it affects how screen reader(s) read the legend text.

Reply:
That's unfortunate, because that means we would have to modify `Choice` to get that effect. What if they were wrapped in a `span` instead of a `div`? Does that make a difference?

Reply:
I'm not sure spans would make a difference; my guess is probably not. An (un)ordered list has an intrinsic relationship to its list items, and screen readers pick up on that. Non-semantic containers like divs and spans don't make that relationship on their own.

As I typed this response, I considered the prospect of adding `role="list"` and `role="listitem"` to the container div, and child divs. StackOverflow seems to support this possibility, and could give us the binding relationship we are looking for: https://stackoverflow.com/a/11131648

Reply:
I will need to research further for the right combination of roles, tabindex, and placement for all screen reader combinations. There is promise in my previous comment, but it will be an enhancement when I have a bit more time to research across the major assistive devices.

Reply:
@pwolfert Is this the last piece preventing a merge?

Sounds like there are two threads here:

1. Whether `MonthPicker` should render the "label" (via a `legend`) for these checkboxes, like `ChoiceList`
2. and the HTML semantics related to the root container, checkbox containers, and label (if the answer to 1. is yes)

for 1, I'm struggling to think of a case where you wouldn't want a label for these fields. I'd be curious to see this component in the context of how y'all are using it in Tax Tools. I assume it has a label asking the user something like "Select the months you lived in X"?